src/gui/sensor/qgssensorwidget.h

```diff
@@ -0,0 +1,161 @@
+/***************************************************************************
+    qgssensorwidget.h
+    ---------------------
+    begin                : March 2023
+    copyright            : (C) 2023 by Mathieu Pellerin
+    email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSSENSORWIDGET_H
+#define QGSSENSORWIDGET_H
+
+#include "ui_widget_tcpsocketsensor.h"
+#include "ui_widget_udpsocketsensor.h"
+#include "ui_widget_serialportsensor.h"
+
+#include "qgis_sip.h"
+#include "qgis_gui.h"
+#include "qgsabstractsensor.h"
+
+#include <QWidget>
+
+/**
+ * \ingroup gui
+ * \class QgsAbstractSensorWidget
+ * \brief Base class for widgets which allow control over the properties of sensors.
+ * \since QGIS 3.32
+ */
+class GUI_EXPORT QgsAbstractSensorWidget : public QWidget
+{
+
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Constructor for QgsAbstractSensorWidget.
+     * \param parent parent widget
+     */
+    QgsAbstractSensorWidget( QWidget *parent SIP_TRANSFERTHIS = nullptr );
+
+    /**
+     * Creates a new sensor matching the settings defined in the widget.
+     */
+    virtual QgsAbstractSensor *createSensor() = 0 SIP_FACTORY;
+
+    /**
+     * Updates an existing sensor to match the settings defined in the widget. If
+     * TRUE is returned, the \a sensor was successfully updated.
+     *
+     * If FALSE is returned, then the widget could not successfully update
+     * the \a sensor.
```

Comment:
This function is supposed to return false only if the underlying sensor type is wrong or also if for example the entered values are out of range?

Reply:
Only returns false is the underlying sensor type can't be updated, it will not care for out of range value here as the UI itself won't allow for it (e.g. minimum / maximum values set on port spinbox).